osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -88,6 +90,30 @@ public BeatmapOffsetControl()
             };
         }
 
+        public class OffsetSliderBar : PlayerSliderBar<double>
+        {
+            protected override Drawable CreateControl() => new CustomSliderBar();
+
+            protected class CustomSliderBar : SliderBar
+            {
+                protected override LocalisableString GetTooltipText(double value)
+                {
+                    return value == 0
+                        ? new TranslatableString("_", @"{0} ms", base.GetTooltipText(value))
+                        : new TranslatableString("_", @"{0} ms {1}", base.GetTooltipText(value), getEarlyLateText(value));
+                }
+
+                private LocalisableString getEarlyLateText(double value)
+                {
+                    Debug.Assert(value != 0);
+
+                    return value > 0
+                        ? BeatmapOffsetControlStrings.HitObjectsAppearLater
+                        : BeatmapOffsetControlStrings.HitObjectsAppearEarlier;
```

Comment:
Now that I look at this again, isn't this actually flipped? if a negative offset is set, then the objects hit time would be *later* than the beat in the track (since a negative offset makes the gameplay clock behind the track), and vice versa.
```suggestion
                    return value > 0
                        ? BeatmapOffsetControlStrings.HitObjectsAppearEarlier
                        : BeatmapOffsetControlStrings.HitObjectsAppearLater;
```

Reply:
Means the calibration is also probably wrong. I haven't been able to test properly, but seems correct. On stable this is the reason the beatmap offset is inverse of the UO. Not yet sure which part to change here. I have always seen this offset as negative meaning hitobjects get moved back and I'd say that's the more common way of thinking about it.

Reply:
Unsure how to feel about two different "offset" settings being opposite to each other. I agree with negative offset meaning "early hitobject appearance", and I think it would also be better to enforce that on both universal audio offset and beatmap offset.

In fact, the current universal audio offset setting in osu!(lazer) is labelled "Audio Offset", but what it's affecting is the gameplay clock, which would mean applying inversely to the audio track (i.e. `500ms` means track is technically -500ms from gameplay clock, not 500ms).